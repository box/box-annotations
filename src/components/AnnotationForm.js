/**
 * @flow
 * @file Wraps a component in an IntlProvider
 * @author Box
 */

import React from 'react';
import noop from 'lodash/noop';

import ApprovalCommentForm from '../../third-party/components/ApprovalCommentForm';
import Internationalize from './Internationalize';
import withFocus from './withFocus';

type Props = {
    createAnnotation: Function,
    onCancel: Function,
    language?: string,
    messages?: StringMap,
    placeholderText: string
};

const DUMMY_USER = {};
const IS_EDITING = true;
const KEEP_INPUT_OPEN = true;

const AnnotationForm = ({ createAnnotation, onCancel, placeholderText, language, messages: intlMessages }: Props) => (
    <Internationalize language={language} messages={intlMessages}>
        <ApprovalCommentForm
            className='ba-annotation-input-container'
            user={DUMMY_USER}
            isOpen={KEEP_INPUT_OPEN}
            isEditing={IS_EDITING}
            createComment={createAnnotation}
            onCancel={onCancel}
            onSubmit={noop}
            onFocus={noop}
            getAvatarUrl={() => {}}
            tagged_message={placeholderText}
        />
    </Internationalize>
);

export { AnnotationForm as AnnotationFormComponent };
export default withFocus(AnnotationForm);
