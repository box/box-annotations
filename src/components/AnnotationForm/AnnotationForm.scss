.ba {
    .bcs-comment-input-container {
        border-top: none;
    }

    .bcs-comment-input-form-container {
        margin-right: 0;

        form {
            margin: 0;
        }
    }

    .bcs-comment-input {
        margin-top: 0;

        ~ .bcs-comment-input-controls .btn {
            margin-bottom: 0;
        }
    }
}
