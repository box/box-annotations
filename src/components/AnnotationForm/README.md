### Description
Component to represent the wrapper around the ApprovalCommentForm component

### Examples
```jsx
const AnnotationFormExample = require('examples').AnnotationFormExample;

<AnnotationFormExample />
```
