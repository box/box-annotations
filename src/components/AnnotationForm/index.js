// @flow
export { default } from './AnnotationForm';
