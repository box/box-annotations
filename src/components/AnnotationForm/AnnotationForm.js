// @flow
import React from 'react';
import noop from 'lodash/noop';

import ApprovalCommentForm from '../../../third-party/components/ApprovalCommentForm';
import Internationalize from '../Internationalize';
import withFocus from '../withFocus';

import './AnnotationForm.scss';

type Props = {
    onCreate: Function,
    onCancel: Function,
    language?: string,
    messages?: StringMap,
    placeholderText?: string
};

type State = {
    isOpen: boolean
};

const DUMMY_USER = {};
const IS_EDITING = true;

class AnnotationForm extends React.Component<Props, State> {
    state = {
        isOpen: true
    };

    componentWillMount() {
        this.setState({ isOpen: true });
    }

    componentWillUnmount() {
        this.setState({ isOpen: false });
    }

    render() {
        const { onCreate, onCancel, placeholderText, language, messages: intlMessages } = this.props;
        const { isOpen } = this.state;

        return (
            <Internationalize language={language} messages={intlMessages}>
                <ApprovalCommentForm
                    className='ba-annotation-input-container'
                    user={DUMMY_USER}
                    isOpen={isOpen}
                    isEditing={IS_EDITING}
                    createComment={onCreate}
                    onCancel={onCancel}
                    onSubmit={noop}
                    onFocus={noop}
                    getAvatarUrl={noop}
                    tagged_message={placeholderText}
                />
            </Internationalize>
        );
    }
}

export { AnnotationForm as AnnotationFormComponent };
export default withFocus(AnnotationForm);
