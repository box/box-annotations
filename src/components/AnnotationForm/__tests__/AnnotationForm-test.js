import * as React from 'react';
import { shallow } from 'enzyme';

import AnnotationForm from '../AnnotationForm';

const onCreate = jest.fn();
const onCancel = jest.fn();

describe('components/AnnotationForm', () => {
    test('should correctly render a list of annotation', () => {
        const wrapper = shallow(<AnnotationForm onCreate={onCreate} onCancel={onCancel} />).dive();
        expect(wrapper).toMatchSnapshot();
    });
});
