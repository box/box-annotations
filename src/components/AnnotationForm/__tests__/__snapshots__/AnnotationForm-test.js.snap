// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AnnotationForm should correctly render a list of annotation 1`] = `
<Internationalize>
  <ApprovalCommentForm
    className="ba-annotation-input-container"
    createComment={[MockFunction]}
    getAvatarUrl={[Function]}
    isEditing={true}
    isOpen={true}
    onCancel={[MockFunction]}
    onFocus={[Function]}
    onSubmit={[Function]}
    user={Object {}}
  />
</Internationalize>
`;
